### DISCUSSION TITLE
Title bar button icons not showing

### DISCUSSION THREAD

**Discussion by opensourcenebula** [jdan/98.css#113] [Q&A] [answered]
I wasn't sure where to put this, but when linking the stylesheet from unpkg it works fine. So I assumed its not an issue with the code but something I was doing.

When attempting to create a window, the title bar button icons just don't appear. Only the icons. The buttons and every other part of the title bar appear just fine. When I switch to the unpkg link, it's perfectly fine, but the moment I switch back the buttons break again. They're the only thing affected this way.

![image](https://user-images.githubusercontent.com/39895112/115947976-e3430c00-a490-11eb-8b7f-d33252fc0b48.png)
Using `<link rel="stylesheet" href="https://unpkg.com/98.css" />`

![image](https://user-images.githubusercontent.com/39895112/115948002-15ed0480-a491-11eb-9a1e-3f0aaad4043b.png)
Using `<link href="/style.css" rel="stylesheet" type="text/css" media="all">`

I'm not sure what could be causing this. I will admit that I'm very rusty with both CSS and HTML and didn't have the best knowledge of either to begin with. I'm kinda blindly fumbling my way through right now. I do have everything that (I think) is necessary for the code to work properly. Being styles.css and the fonts and icon folder, since those are the only things that seem to be referenced in the CSS.

I would prefer to host the CSS myself just because I'm going to use it to build a complete looking desktop and would like to make additions right here as I need them, if I need them. I am hosting through Neocities if that has any effect on anything, but I don't think it would since the unpkg link works fine.

**Comment by opensourcenebula** [comment]
Update: Maybe a hacky solution, maybe a genuine solution to my problem, but while working on other items I kept looking back and noticing that svg-load wasn't being considered valid by Neocities's editor. I changed it from `svg-load("./icon/minimize.svg")` to `url("./icon/minimize.svg")` and that made it work... okay?

![image](https://user-images.githubusercontent.com/39895112/115953581-f7e3cc00-a4b1-11eb-8700-10b6180ee9be.png)

I don't know if this is just a problem with my choice of host or what.

**Comment by jdan** [comment, marked as answer]
Hey, thanks for the good question and repro steps! `style.css` needs to be built with `npm run build` (where it becomes `dist/98.css` which unpkg links to).

Wanna give that a whirl and let me know if that fixes the icons?

**Reply by opensourcenebula** [reply]
I work in IT. I know how it feels to not have enough information. I try my best to do better for others :')

I didn't realize that npm was a needed step, so it was a my bad. Though now I'm having trouble with npm so I can't say for sure if that's going to work. I've never used it before.

If I can ask for help on that too, (;;) it got in about a second before spitting out this:
```
Error: Cannot find module 'dedent'
Require stack:
- C:\Users\Neptune\Downloads\98.css-main\98.css-main\build.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\Users\Neptune\Downloads\98.css-main\98.css-main\build.js:2:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\Neptune\\Downloads\\98.css-main\\98.css-main\\build.js'
  ]
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! 98.css@0.1.17 build: `node build.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the 98.css@0.1.17 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Neptune\AppData\Roaming\npm-cache\_logs\2021-04-24T17_18_28_167Z-debug.log
```

Heres the log too:
[2021-04-24T17_18_28_167Z-debug.log](https://github.com/jdan/98.css/files/6370223/2021-04-24T17_18_28_167Z-debug.log)

**Reply by jdan** [reply]
You may need to install dependencies with `npm install` (or `npm i` for short) before running any `npm run __` commands.

**Reply by opensourcenebula** [reply]
That worked! Thank you so much for helping with this!

**Reply by jdan** [reply]
So very welcome. Best of luck with whatever you're building (& keep me posted!)
